poo/abstraction/en/Abstraction.md

```diff
@@ -0,0 +1,131 @@
+# Abstraction
+
+**Abstraction** is the idea of dividing a class into several parts, to be able to share (via a parameter passed in a method for example) **only one** of these parts, rather than the object entirely. To understand the utility of this principle and how to apply it, take for example a class `Human` since this is something that we know well.
+
+
+```java
+
+class Human {
+
+    int air;
+    int satiety;
+    float energy;
+
+
+    void breathe() {
+        air += 10;
+    }
+
+    void eat() {
+        satiety += 5;
+    }
+
+    void move() {
+        System.out.println("A human wants to move!");
+    }
+
+    void consumeEnergy(float value) {
+        this.energie -= value;
+    }
+}
+```
+
+Now, imagine that we have a class `Hiking` that allows us to organise a walk. Something like:
+
+```java
+
+class Hiking {
+
+    List<Human> participants;
+
+    public Hiking() { this.participants = new ArrayList<>(); }
+
+    void addParticipant(Human participant) { this.participants.add(participant); }
+
+    void moveAll() {
+        for(Human human : participants) {
+            human.move();
+            human.consumeEnergy(3);
+        }
+    }
+}
+```
+
+So far so good, we have what we want. This code works perfectly well, except that, in reality, it contains two major problems.
+
+1. **We restraint the hiking of the objects of type Human**, whereas anything that can **walk** and **be exhausted** could take a hike. It’s not really the kind to forbid to elephants and to tractors to participate :(

```

Comment:
Why is there a `(` here?

Reply:
This is a smiley, the same of the fr version
> 1. On restreint la randonnée aux objets de type Human, alors que n'importe quoi qui peut marcher et s'épuiser pourrait faire une randonnée. C'est vraiment pas gentil d'interdire aux éléphants et aux tracteurs de participer :(

Reply:
Oh okay 😂. On my phone I just saw a `(`

Reply:
No problem! 😄